### PR TITLE
ArC fixes for spec compatibility

### PR DESCRIPTION
**Pull request by Ladicek**
Related to #28558

- scan `package-info.class` for `@Vetoed`
  - including a small fix to `ArcTestContainer` to also index `package-info.class` of all classes in the test bean archive
- fix dynamic lookup of array types
  - applies when required _array_ type during dynamic lookup is represented as a `Class`, while the bean _array_ types are instantiated as `GenericArrayType`
  - this scenario was already handled in the Weld codebase, so this is just a direct copy
- fix `BeanManager.getReference()` implementation
  - was missing one check, which is copied directly from Weld again
- fix treatment of primitive wrapper type producers that produce `null`
  - the zero value must be injected into primitive injection points
- fix treatment of injection points with `@Named` without value
  - field injection points with `@Named` without value should use `@Named(fieldName)` as the qualifier
  - other such injection points are a definition error

To review, it's probably best to look at each commit in isolation. Tests are added, too.

### CODE REVIEW

**Comment by gsmet** [comment]
The failures are not related.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java [hunk 1]
```diff
@@ -706,6 +714,46 @@ static boolean isPrimitiveClassName(String className) {
...
 
+    static boolean isPrimitiveWrapperType(Type type) {
+        if (type.kind() == Kind.CLASS) {
+            return PRIMITIVE_WRAPPERS.contains(type.name());
+        }
+        return false;
+    }
+
+    /**
+     * Emits a bytecode instruction to load the zero value of given {@code primitive} type
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java:725]: I think that both CDI and JLS specs talk about "default values" and not "zero values" but that's a pretty useless nitpicking from my side :D

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Named("foo")
+    @Dependent
+    static class Foo {
+    }
+
+    @Dependent
+    static class Consumer {
+        @Inject
+        Consumer(@Named Foo foo) {
```
Comment (mkouba) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java:40]: In theory, we could be a bit more lenient and if a param name is available then just use it instead of throwing a definition exception. On the other hand, `@Named` at injection points should not be rarely used. So it's probably not worth the effort...

**Review by mkouba — APPROVED** [review]
Cool! Thanks!

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java (hunk 2 above)
Comment (Ladicek) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java:40]: True. The spec says it's a definition error, and while we could improve on this, I didn't feel like spending more time on this particular corner case :-)

Comment (mkouba) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java:40]: :+1: when this rule was created it was not possible to store the param names in the class file ;-)

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java (hunk 1 above)
Comment (Ladicek) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java:725]: Correct. I wanted to rebase this PR anyway after #30544 is merged, so I'll fix this :-)

**Review by manovotn — APPROVED** [review]
Looks good, thanks!

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java (hunk 1 above)
Comment (Ladicek) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java]: Done.

**Comment by Ladicek** [comment]
Rebased (and addressed the nitpick :laughing:).
